x-pack/plugins/reporting/server/export_types/csv/execute_job.ts

```diff
@@ -4,79 +4,44 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { Crypto } from '@elastic/node-crypto';
-import { i18n } from '@kbn/i18n';
-import Hapi from 'hapi';
-import { KibanaRequest } from '../../../../../../src/core/server';
 import { CONTENT_TYPE_CSV, CSV_JOB_TYPE } from '../../../common/constants';
-import { cryptoFactory, LevelLogger } from '../../lib';
 import { RunTaskFn, RunTaskFnFactory } from '../../types';
+import { decryptJobHeaders } from '../common';
 import { createGenerateCsv } from './generate_csv';
 import { TaskPayloadCSV } from './types';
 
-const getRequest = async (headers: string | undefined, crypto: Crypto, logger: LevelLogger) => {
-  const decryptHeaders = async () => {
-    try {
-      if (typeof headers !== 'string') {
-        throw new Error(
-          i18n.translate(
-            'xpack.reporting.exportTypes.csv.executeJob.missingJobHeadersErrorMessage',
-            {
-              defaultMessage: 'Job headers are missing',
-            }
-          )
-        );
-      }
-      return await crypto.decrypt(headers);
-    } catch (err) {
-      logger.error(err);
-      throw new Error(
-          i18n.translate(
-            'xpack.reporting.exportTypes.csv.executeJob.failedToDecryptReportJobDataErrorMessage',
-            {
-              defaultMessage: 'Failed to decrypt report job data. Please ensure that {encryptionKey} is set and re-generate this report. {err}',
-              values: { encryptionKey: 'xpack.reporting.encryptionKey', err: err.toString() },
-            }
-          )
-        ); // prettier-ignore
-    }
-  };
-
-  return KibanaRequest.from({
```

Comment:
We now have the `reporting.getFakeRequest` utility that generates a fake request along with the space ID it needs